tox.ini

```diff
@@ -1,18 +1,16 @@
 [tox]
-envlist = checks,licenses,py{36,37,38,39,310}-{unittest,integration}
+envlist = checks,py{36,37,38,39,310}-{unittest,integration}
+isolated_build = True
 
 [testenv]
 passenv = HOME
-deps =
-    -rdev-requirements.txt
 sitepackages = False
+allowlist_externals = poetry
+commands_pre =
+		pip install poetry
```

Comment:
This seems wrong, we either use the system's poetry (through allowlist_externals) or we install it in `deps`. I recommend using the system's.

Reply:
when I leave only `allowlist_externals = poetry`, I get an error:

`py311 run-test-pre: commands[0] | poetry install --all-extras
ERROR: InvocationError for command could not find executable poetry`

When I add poetry to deps, it works. But I've seen other `tox.ini`s, where poetry is only in `allowlist_externals`. Am I missing something here?

Reply:
That's because you haven't installed poetry on your machine, `dnf install poetry` and it should work.

Reply:
The poetry folks recommend not installing poetry in the same virtualenv as the application you are managing, that's why.